elide-core/src/main/java/com/yahoo/elide/parsers/expression/PermissionToFilterExpressionVisitor.java

```diff
@@ -39,7 +42,17 @@
             return null;
         }
     };
-    public static final FilterExpression USER_CHECK_EXPRESSION = new FilterExpression() {
+
+    /**
+     * A FilterExpression representing userCheck.
+     */
+    public class UserCheckExpression implements FilterExpression {
```

Comment:
I'm wondering if we could keep the original USER_CHECK_EXPRESSION and do the following edits (see below):